lambda/ensureThingExists.ts

```diff
@@ -0,0 +1,15 @@
+import { CreateThingCommand, type IoTClient } from '@aws-sdk/client-iot'
+import { thingExists } from '../nordicNRPlus/thingExists.ts'
+
+export const ensureThingExists =
+	(iotClient: IoTClient) =>
+	async (thingName: string): Promise<void> => {
+		if ((await thingExists(iotClient, thingName)) === false) {
+			await iotClient.send(
+				new CreateThingCommand({
+					thingName,
+					thingTypeName: 'nordic-nrplus',
+				}),
+			)
+		}
+	}
```

Comment:
```suggestion

export const ensureThingExists =
	(iotClient: IoTClient, thingExists: EnsureThingFn) => {
	
	const e = thingExists(iotClient)
	
	return async (thingName: string): Promise<void> => {
	    if (await e(thingName)) return
		await iotClient.send(
			new CreateThingCommand({
				thingName,
				thingTypeName: 'nordic-nrplus',
			}),
		)
	}
	
	}
```

1. Do not inject dependencies on every call
2. Keep code left